src/bug-report/BugReportTool/BugReportTool/Main.cpp

```diff
@@ -0,0 +1,213 @@
+    {
+        for (auto it : val.GetArray())
+        {
+            hideByXPath(it, xpathArray, p);    
+        }
+
+        return;
+    }
+
+    if (p == xpathArray.size() - 1)
+    {
+        if (val.ValueType() == JsonValueType::Object)
+        {
+            auto obj = val.GetObjectW();
+            if (obj.HasKey(xpathArray[p]))
+            {
+                auto privateDatavalue = JsonValue::CreateStringValue(L"<private_data>");
+                obj.SetNamedValue(xpathArray[p], privateDatavalue);
+            }
+        }
+        
+        return;
+    }
+
+    if (val.ValueType() == JsonValueType::Object)
+    {
+        IJsonValue newVal;
+        try
+        {
+            newVal = val.GetObjectW().GetNamedValue(xpathArray[p]);
+        }
+        catch (...)
+        {
+            return;
+        }
+        
+        hideByXPath(newVal, xpathArray, p + 1);    
+    }
+}
+
+void hideForFile(path dir, wstring relativePath)
+{
+    wstring jsonString, tmp;
+    dir = dir.append(relativePath);
+    wifstream file(dir);
+    while (std::getline(file, tmp))
+    {
+        jsonString += tmp;
+    }
+
+    JsonValue jValue = NULL;
+    if (!JsonValue::TryParse(jsonString, jValue))
+    {
+        wprintf(L"Can not parse file %s", dir.c_str());
+        return;
+    }
+
+    for (auto xpath : escapeInfo[relativePath])
+    {
+        vector<wstring> xpathArray = getXpathArray(xpath);
+        hideByXPath(jValue, xpathArray, 0);
+    }
+
+    jsonString = jValue.Stringify();
+    wofstream out(dir);
+    out << jsonString;
+}
+
+bool del(wstring path)
+{
+    error_code err;
+    remove_all(path, err);
+    if (err.value() != 0)
+    {
+        wprintf_s(L"Can not delete %s. Error code: %d", path.c_str(), err.value());
+        return false;
+    }
+
+    return true;
+}
+
+void hideUserPrivateInfo(filesystem::path dir) 
+{
+    // Replace data in json files
+    for (auto& it : escapeInfo)
+    {
+        hideForFile(dir, it.first);
+    }
+
+    // delete files
+    for (auto it : filesToDelete)
+    {
+        del(it);
+    }
+}
+
+int wmain(int argc, wchar_t* argv[], wchar_t*)
+{
+    // Get path to save zip
+    wstring saveZipPath;
+    if (argc > 1)
+    {
+        saveZipPath = argv[1];
+    }
+    else
+    {
+        wchar_t buffer[MAX_PATH];
+        if (SHGetSpecialFolderPath(HWND_DESKTOP, buffer, CSIDL_DESKTOP, FALSE))
+        {
+            saveZipPath = buffer;
+        }
+        else
+        {
+            printf("Can not retrieve a desktop path. Error code: %d", GetLastError());
+        }
+    }
+
+    auto powerToys = PTSettingsHelper::get_root_save_folder_location();
+    
+    // Copy to a temp folder
+    auto tmpDir = temp_directory_path();
+    tmpDir = tmpDir.append("PowerToys\\");
+    powerToys = powerToys + L"\\";
+    if (!del(tmpDir))
+    {
+        return 1;
+    }
+
+    try
+    {
+        copy(powerToys, tmpDir, copy_options::recursive);
+    }
+    catch (std::bad_alloc err)
```

Comment:
there's no reason to handle bad_alloc separately from catch(...)

Reply:
Why? Potentially, we might get extra information about exception. Am I missing something?

Reply:
bad_alloc is thrown when there's no memory left in the system, we can't do any meaningful analysis with it (other than we're leaking memory).